New dependency version: VTK versions 9.4 and 9.5

Library that has a new version: VTK

Link to release notes: 
 * 9.4 https://docs.vtk.org/en/latest/release_details/9.4.html
 * 9.5 https://docs.vtk.org/en/latest/release_details/9.5.html

**TODO:**
- [ ]  Build a docker image containing the new dependency version - see the [Portability Testing Guide](https://github.com/Chaste/developer_wiki/wiki/Portability-Testing-Guide) for how to do this.
- [x]  Create a branch off `develop` - you could name it, for example, `support-petsc-322`.
- [ ]  Add the new docker image to the [portability workflow](https://github.com/Chaste/Chaste/blob/develop/.github/workflows/portability.yml) on the new branch.
- [x]  Create a PR with the new branch and see if it builds and passes all tests.
- [x]  Do any necessary fixes.
  - [x] 9.4
  - [x] 9.5
- [ ]  Add permanently to testing rotation i.e. either replace an old docker image in the [portability workflow](https://github.com/Chaste/Chaste/blob/develop/.github/workflows/portability.yml) with the new image, or append the new image to the existing list of portability images. For example, if there's an old dependency version that is being retired, the new docker image can probably replace the old one.
- [ ]  Update the portability versions in the [Portability Testing Guide](https://github.com/Chaste/developer_wiki/wiki/Portability-Testing-Guide).
- [ ]  Update the [supported dependency versions](https://chaste.github.io/docs/installguides/dependency-versions/) page.

See #500 

VTK 9.5.2 causes a failure in TestAirwayGenerator, with quite a few failures including e.g.

```
366: /home/fcooper/GitRepos/Chaste/Chaste_sandbox/lung/test/airway_generation/TestAirwayGenerator.hpp:521: Error: Expected (static_cast<unsigned long>(generator.GetAirwayTree()->GetNumberOfPoints()) == 97ul), found (121 != 97)
```

This is very likely to be VTK-related but haven't yet dug into the details.

--- a/lung/src/airway_generation/AirwayGenerator.cpp
+++ b/lung/src/airway_generation/AirwayGenerator.cpp
@@ -54,15 +54,14 @@ OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #if ((VTK_MAJOR_VERSION >= 5 && VTK_MINOR_VERSION >= 6) || VTK_MAJOR_VERSION >= 6)
 
-#include "vtkGeometryFilter.h"
 #include "vtkDoubleArray.h"
+#include <vtkClipPolyData.h>
 #include "vtkPlane.h"
 #include "vtkPoints.h"
 #include "vtkPointData.h"
 #include "vtkPolyVertex.h"
 #include "vtkMassProperties.h"
 #include "vtkMath.h"
-#include "vtkTableBasedClipDataSet.h"
 #include "vtkCellArray.h"
 #include "vtkLine.h"
 #include "vtkUnsignedIntArray.h"
@@ -235,26 +234,19 @@ vtkSmartPointer<vtkPolyData> AirwayGenerator::SplitPointCloud(vtkSmartPointer<vt
                                                               double rOrigin[3],
                                                               bool insideOut)
 {
-    vtkSmartPointer<vtkPlane> clip_plane = vtkSmartPointer<vtkPlane>::New();
+    auto clip_plane = vtkSmartPointer<vtkPlane>::New();
     clip_plane->SetNormal(rNormal);
     clip_plane->SetOrigin(rOrigin);
 
-    vtkSmartPointer<vtkTableBasedClipDataSet > clipper = vtkSmartPointer<vtkTableBasedClipDataSet >::New();
-#if VTK_MAJOR_VERSION >= 6
+    auto clipper = vtkSmartPointer<vtkClipPolyData>::New();
     clipper->SetInputData(pointCloud);
-#else
-    clipper->SetInputConnection(pointCloud->GetProducerPort());
-#endif
     clipper->SetClipFunction(clip_plane);
-    clipper->GenerateClippedOutputOff();
+    clipper->SetGenerateClipScalars(1);
     clipper->SetInsideOut(insideOut);
+    clipper->GenerateClippedOutputOff();
     clipper->Update();
 
-    vtkSmartPointer<vtkGeometryFilter> filter = vtkSmartPointer<vtkGeometryFilter>::New();
-    filter->SetInputConnection(clipper->GetOutputPort());
-    filter->Update();
-
-    return filter->GetOutput();
+    return clipper->GetOutput();
 }
 
 void AirwayGenerator::AddInitialApex(double rStartLocation[3],

